corehq/messaging/scheduling/static/scheduling/js/conditional_alert_main.js

```diff
@@ -34,5 +34,34 @@ hqDefine("scheduling/js/conditional_alert_main", [
         // setup tab
         basicInformation.setRuleTabVisibility();
         $("#conditional-alert-basic-info-panel").koApplyBindings(basicInformation);
+
+        $("#conditional-alert-save-btn").click(function () {
+            var newRuleName = $("#id_conditional-alert-name").val();
+            if (!name) {
+                $("#conditional-alert-form").submit();
+            } else if (name !== newRuleName) {
+                var caseType = $(
+                    "#select2-id_criteria-case_type-container"
+                ).text();
+                var caseCountUrl = $("#conditional-alert-case-count-url").attr(
+                    "url"
+                );
+                $.ajax({
+                    type: "GET",
+                    url: caseCountUrl,
+                    data: { case_type: caseType },
+                    success: function (data) {
+                        $("#case-count").text(data["case_count"]);
+                    },
+                });
+                $("#conditional-alert-warning-modal").modal("show");
```

Comment:
I'd pull this inside the success function, rather than popping up the modal and displaying an incomplete warning with the number of cases missing.